src/callbacks_step/save_restart_dg.jl

```diff
@@ -178,7 +180,7 @@ function load_restart_file(mesh::ParallelTreeMesh, equations, dg::DG, cache, res
 
       # Read variable
       println("Reading variables_$v ($name)...")
-      u[v, .., :] = MPI.Scatterv(read(file["variables_$v"]), node_counts, mpi_root(), mpi_comm())
+      MPI.Scatterv!(MPI.VBuffer(read(file["variables_$v"]), node_counts), @view(u[v, .., :]), mpi_root(), mpi_comm())
```

Comment:
Can you split this into at least 2 lines please to make it easier to understand what's going on here? At least the `read(..)` should be on a separate line such that new users are not overwhelmed by the complexity of this single-line statement.